emscripten.py

```diff
@@ -1491,75 +1497,121 @@ def float_coerce(s):
 
-    if not settings.get('SIDE_MODULE'):
-      funcs_js.append('''
-Runtime.stackAlloc = Module['stackAlloc'];
-Runtime.stackSave = Module['stackSave'];
-Runtime.stackRestore = Module['stackRestore'];
-Runtime.establishStackSpace = Module['establishStackSpace'];
-''')
-      if settings['SAFE_HEAP']:
-        funcs_js.append('''
-Runtime.setDynamicTop = Module['setDynamicTop'];
-''')
+  if not settings['RELOCATABLE']:
+    overrides += ['setTempRet0', 'getTempRet0']
 
-    if not settings['RELOCATABLE']:
-      funcs_js.append('''
-Runtime.setTempRet0 = Module['setTempRet0'];
-Runtime.getTempRet0 = Module['getTempRet0'];
-''')
+  lines = ["Runtime.{0} = Module['{0}'];".format(func) for func in overrides]
+  return '\n'.join(lines)
 
-    # Set function table masks
-    masks = {}
-    max_mask = 0
-    for sig, table in function_table_data.iteritems():
-      mask = table.count(',')
-      masks[sig] = str(mask)
-      max_mask = max(mask, max_mask)
-    def function_table_maskize(js, masks):
-      def fix(m):
-        sig = m.groups(0)[0]
-        return masks[sig]
-      return re.sub(r'{{{ FTM_([\w\d_$]+) }}}', fix, js) # masks[m.groups(0)[0]]
-    for i in range(len(funcs_js)): # in-place as this can be large
-      funcs_js[i] = function_table_maskize(funcs_js[i], masks)
-
-    if settings['SIDE_MODULE']:
-      funcs_js.append('''
-Runtime.registerFunctions(%(sigs)s, Module);
-''' % { 'sigs': str(map(str, function_table_data.keys())) })
 
-    for i in range(len(funcs_js)): # do this loop carefully to save memory
-      if WINDOWS: funcs_js[i] = funcs_js[i].replace('\r\n', '\n') # Normalize to UNIX line endings, otherwise writing to text file will duplicate \r\n to \r\r\n!
-      outfile.write(funcs_js[i])
-    funcs_js = None
+def create_first_in_asm(settings):
+  first_in_asm = ''
+  if settings['SPLIT_MEMORY']:
+    if not settings['SAFE_SPLIT_MEMORY']:
+      first_in_asm += ''.join([make_get_set(info) for info in HEAP_TYPE_INFOS]) + '\n'
+    first_in_asm += 'buffer = new ArrayBuffer(32); // fake\n'
+  return first_in_asm
 
-    if WINDOWS: post = post.replace('\r\n', '\n') # Normalize to UNIX line endings, otherwise writing to text file will duplicate \r\n to \r\r\n!
-    outfile.write(post)
 
-    if DEBUG:
-      logging.debug('  emscript: python processing: finalize took %s seconds' % (time.time() - t))
+def make_get_set(info):
+  access = ('HEAP{name}s[ptr >> SPLIT_MEMORY_BITS][(ptr & SPLIT_MEMORY_MASK) >> {shift}]'
+            .format(name=info.short_name, shift=info.shift_amount))
+  # TODO: fround when present for Float32
+  return '''
+function get{short}(ptr) {{
+  ptr = ptr | 0;
+  return {coerced_access};
+}}
+function set{short}(ptr, value) {{
+  ptr = ptr | 0;
+  value = {coerced_value};
+  {access} = value;
+}}'''.format(
+    short=info.short_name,
+    coerced_value=info.coerce('value'),
+    access=access,
+    coerced_access=info.coerce(access))
+
+
+def create_memory_views(settings):
```

Comment:
Done in https://github.com/kripken/emscripten/pull/5150/commits/827fbcaa5dc1ea06eabee89452672e1722e8c500